src/applications/vaos/utils/data.js

```diff
@@ -183,20 +184,17 @@ export function transformFormToAppointment(state) {
   const data = getFormData(state);
   const clinic = getChosenClinicInfo(state);
   const facility = getChosenFacilityInfo(state);
-  const slot = data.calendarData.selectedDates[0];
+  const slot = getChosenSlot(state);
   const purpose = getUserMessage(data);
-  const appointmentLength = parseInt(
-    state.newAppointment.appointmentLength,
-    10,
-  );
+  const appointmentLength = moment(slot.end).diff(slot.start, 'minutes');
 
   return {
     appointmentType: getTypeOfCare(data).name,
     clinic,
     // These times are a lie, they're actually in local time, but the upstream
     // service expects the 0 offset.
     desiredDate: `${data.preferredDate}T00:00:00+00:00`,
-    dateTime: moment(slot.datetime).format('YYYY-MM-DD[T]HH:mm:ss[+00:00]'),
+    dateTime: `${slot.start}+00:00`,
```

Comment:
Need to verify this works, since it may include a `.000` piece that wasn't there before.